### PR TITLE
Generation of the schema looks to be deterministic

### PR DESCRIPTION
**Pull request by epii-1**
(Hopefully) managed to pinpoint the non-deterministic behavior of the schema generation down to 3 specific rows and corrected these while possible ever so slightly improving performance.

There is still a chance the output is somewhat random, because of other reasons (OS etc. as mentioned elsewhere).

I don't really know who to tag for this one either. So I'll leave it here and mention it in the issue thread.

### CODE REVIEW

**Comment by hartig** [comment]
Thanks Leif!

@rcapshaw can you please take a look at this PR.

**Comment by rcapshaw** [comment]
It looks like this works, the root cause seems to be the use of set(...) to copy the keys from the various maps. That being said, wrapping things in `list()` only works if the underlying structures are consistently ordered. It looks like they are; keys seem to be ordered by insertion and the library seems to read the input files consistently. We unfortunately can't skip this step since it acts as a shallow copy, and we modify the underlying map as we traverse its keys.

So, in short, looks good, but let's keep an eye on this.

**Comment by hartig** [comment]
Riley, thanks for checking and for merging.
